src/asm/x86/dist/hbd.rs

```diff
@@ -61,9 +61,14 @@ macro_rules! satd_kernel_hbd_avx2 {
           let input1 = &mut Aligned::<[i32; 8 * 8]>::uninitialized().data[..size * size];
           let input2 = &mut Aligned::<[i32; 8 * 8]>::uninitialized().data[..size * size];
           for y in 0..(sizei) {
-            for x in 0..(sizei) {
-              input1.as_mut_ptr().offset(y * sizei + x).write(src.offset(y * src_stride + x).read() as i32);
-              input2.as_mut_ptr().offset(y * sizei + x).write(dst.offset(y * dst_stride + x).read() as i32);
+            if size == 8 {
```

Comment:
I hope this branch is elided by the compiler.

Reply:
It is, I verified that the compiler sees `size` as constant and elides out any branches. So the 8x8 version has the first branch only, and the 4x4 version has the second branch only.